### PR TITLE
Bo kc issue ui backend

### LINKED ISSUE #2: User opens add account dialog
### User Story
As a User, I want to add accounts.

### Context / Additional Information
Given the user is on the dashboard of the app
### Acceptance Criteria
- [x] - Clicking the add account button, the user is directed to the new account dialog
- [x] - New account dialog contains fields for account name and initial balance
- [x] - New account dialog contains save and cancel functions

---------

Given: The user is on the dashboard of the app
When: The user clicks on the add account button
Then: The new account dialog will appear, including an option for name and starting balance, as well as a save or cancel option.

### LINKED ISSUE #3: User Makes new account
### User Story
As a User, I want to add accounts

### Context / Additional Information
Given the user is at the add account dialog, the required information is entered and, the initial balance is greater than 0
### Acceptance Criteria
- [x] - Clicking the save button, the new account is initialized
- [x] - The new account dialog closes
- [x] - The newly created account is visible from the dashboard

---------

Given: The user is on the add account dialog, and, the two text entry boxes for the name and starting balance are filled.
When: the user presses the save button
Then: A new account will be made, and the new account dialog will close.

### LINKED ISSUE #5: User opens deposits / withdrawals dialog
### User Story
As a Banker, I want to deposit and withdraw money from the account

### Context / Additional Information
Given the user has selected a valid account
### Acceptance Criteria
- [x] - Clicking the deposit/withdraw button, the deposit/withdrawal dialogue pops up
- [x] - The deposit/withdrawal dialog includes options for amount and memorandum (memo)
- [x] - The deposit/withdrawal dialog includes options for save and cancel

------------

Given: the user has selected an account and is on the account screen
When: the user selects the deposit / withdraw button
Then: the deposit /withdraw dialog will pop up, including options for amount and an added note. The save and cancel function are also available here.

### LINKED ISSUE #9: Hard set account value
### User Story
As a BANKER I want to set the balance for an account

### Acceptance Criteria
- [ ] Banker can update the account balance to a specified value
- [ ] Action remembered in transaction history
- [ ] Option to clear history on set value?

### LINKED ISSUE #22: UI: Accounts Page
### User Story
As a user, when I open the app, I want to see the accounts page

### PR DESCRIPTION
**Pull request by RealDogDad**
Connects Create Account UI to Create Account DB
Allows for the creation of multiple Accounts in the DB
The Accounts are now reflected in the Main Dashboard

### CODE REVIEW

**Review by Dylan-Riley — APPROVED** [review]
Checked out branch on my machine and worked as you demoed earlier today. Buttons for withdraw and deposit are still weirdly offset on my screen, I'll add that as something to look into more myself

**Comment by jleeboyd** [comment]
@theanticrumpet  What emulator are you using?

File: app/src/main/java/com/example/family_bank_app/TransactionEntity.java [hunk 1]
```diff
@@ -3,17 +3,21 @@
...
+import androidx.room.ForeignKey;
 import androidx.room.PrimaryKey;
 
 // Store the transactions related to each account, one to many with account Entity
 //https://developer.android.com/training/data-storage/room/relationships
-@Entity
+@Entity(foreignKeys = @ForeignKey(entity = AccountEntity.class,
+        parentColumns = "accountUid",
+        childColumns = "transactionUid",
+        onDelete = ForeignKey.CASCADE))
```
Comment (jleeboyd) [inline app/src/main/java/com/example/family_bank_app/TransactionEntity.java:14]: Are you using this? The same code is already created in the AccountWithTransactions activity.

**Review by jleeboyd — APPROVED** [review]
See comment on Entity code. Otherwise looks good

**Pull request closed without merging**